### PR TITLE
SBRA 447 Bugfix Multi-gpu

### PR DESCRIPTION
**Pull request by cbovalo**
This PR proposes a workaround to avoid creating multiple experiment folders when using several GPUs.
In addition, in this context, all batches are gathered during testing to generate plots.
It also fixes minor bugs and/or warnings.

### CODE REVIEW

File: combustion/gnns/config.py [hunk 1]
```diff
@@ -53,9 +53,23 @@
...
-                    format='%(name)s - %(levelname)s - %(message)s')
+if os.getenv("AI4SIM_EXPERIMENT_PATH") is None:
+    os.environ["AI4SIM_EXPERIMENT_PATH"] = experiment_path
+    os.environ["AI4SIM_LOGS_PATH"] = logs_path
+    os.environ["AI4SIM_ARTIFACTS_PATH"] = artifacts_path
+    os.environ["AI4SIM_PLOTS_PATH"] = plots_path
+elif os.getenv("AI4SIM_EXPERIMENT_PATH") != experiment_path:
+    shutil.rmtree(experiment_path)
+    experiment_path = os.getenv("AI4SIM_EXPERIMENT_PATH")
+    logs_path = os.getenv("AI4SIM_LOGS_PATH")
```
Comment (A669015) [inline combustion/gnns/config.py:64]: What happens is these below env var are not set ? I mean if only AI4SIM_EXPERIMENT_PATH is set and we pass in this branch. The other pathes are set to None ?

File: combustion/gnns/tests/test_config.py [hunk 2]
```diff
@@ -23,18 +28,67 @@ class TestConfig(unittest.TestCase):
...
+        self.assertTrue(os.getenv("AI4SIM_ARTIFACTS_PATH"), config.artifacts_path)
 
     def test_plots_path(self):
         """Test if config creates the correct plots path."""
         self.assertTrue(os.path.exists(config.plots_path))
+        self.assertTrue(os.getenv("AI4SIM_PLOTS_PATH"), config.plots_path)
+
+    def create_env(self, tempdir):
+        """Create a test environment and data test."""
+        os.mkdir(os.path.join(tempdir, "raw_data"))
```
Comment (VTrappler) [inline combustion/gnns/tests/test_config.py:55]: might throw an error if it exists ?

**Review by VTrappler — APPROVED** [review]
there might be missing a few comments, but I guess it is more on the whole repo than on this specific PR

File: combustion/gnns/config.py (hunk 1 above)
Comment (cbovalo) [reply combustion/gnns/config.py:64]: To do it correctly, we should also check if the env var are set. If they are not, the training will fail as all the processes need these paths (all python scripts call config.py).
If they are not set, it means that the code above failed.

**Review by leonicoletti — APPROVED** [review]

**Pull request merged**